modules/gdscript/gdscript.cpp

```diff
@@ -1939,32 +1962,35 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 	for (GDScript *sc : pl) {
 		HashMap<StringName, GDScriptFunction *>::Iterator E = sc->member_functions.find(GDScriptLanguage::get_singleton()->strings._notification);
 		if (E) {
-			Callable::CallError err;
-			E->value->call(this, args, 1, err);
-			if (err.error != Callable::CallError::CALL_OK) {
-				//print error about notification call
-			}
+			Callable::CallError ce;
+			E->value->call(this, args, 1, ce);
+			// No `return`, **do** multi-level calls!
 		}
 	}
 }
 
 String GDScriptInstance::to_string(bool *r_valid) {
-	if (has_method(CoreStringNames::get_singleton()->_to_string)) {
-		Callable::CallError ce;
-		Variant ret = callp(CoreStringNames::get_singleton()->_to_string, nullptr, 0, ce);
-		if (ce.error == Callable::CallError::CALL_OK) {
-			if (ret.get_type() != Variant::STRING) {
-				if (r_valid) {
-					*r_valid = false;
-				}
-				ERR_FAIL_V_MSG(String(), "Wrong type for " + CoreStringNames::get_singleton()->_to_string + ", must be a String.");
+	GDScript *sptr = script.ptr();
+	while (sptr) {
```

Comment:
Is there a reason we can't use `GDScriptInstance::callp()` here? It already appears to do what we'd want  without multi-level calls:
```
while (sptr) {
	HashMap<StringName, GDScriptFunction *>::Iterator E = sptr->member_functions.find(p_method);
	if (E) {
		return E->value->call(this, p_args, p_argcount, r_error);
	}
	sptr = sptr->_base;
}
```

Reply:
This is a small optimization. `has_method()` and `callp()` both look for a method in base classes, but we don't need to look for a method a second time if we've already found it. Also this is consistent with other methods.